### PR TITLE
override functions excluded from FUNCTION_BOOLEAN_PREFIX

### PR DESCRIPTION
**Pull request by PhilipDukhov**
We don't need to ask subclasses to rename superclass methods

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 1]
```diff
@@ -347,11 +348,14 @@ class IdentifierNaming(private val configRules: List<RulesConfig>) : Rule("ident
         // check for methods that return Boolean
         val functionReturnType = node.findChildAfter(VALUE_PARAMETER_LIST, TYPE_REFERENCE)?.text
 
-        // if function has Boolean return type in 99% of cases it is much better to name it with isXXX or hasXXX prefix
-        if (functionReturnType != null && functionReturnType == PrimitiveType.BOOLEAN.typeName.asString()) {
-            if (BOOLEAN_METHOD_PREFIXES.none { functionName.text.startsWith(it) }) {
-                FUNCTION_BOOLEAN_PREFIX.warnAndFix(configRules, emitWarn, isFixMode, functionName.text, functionName.startOffset, functionName) {
-                    // FixMe: add agressive autofix for this
+        // We don't need to ask subclasses to rename superclass methods
+        if (node.treeParent.findChildByType(MODIFIER_LIST)?.findChildByType(OVERRIDE_KEYWORD) == null) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:352]:
```suggestion
        if (node.treeParent.isOverriden()) {
```
We have a special extension function in `org.cqfn.diktat.ruleset.utils`.

**Review by petertrr — CHANGES_REQUESTED** [review]
Thanks for the PR, @PhilipDukhov! Your change seems useful. Could you please fix the CI build (https://github.com/cqfn/diKTat/pull/567/checks?check_run_id=1450878080)?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 1 above)
Comment (PhilipDukhov) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt]: ok, it's done
on my local machine I wasn't been able to build current version. have an error `Redeclaration: AstNodeUtilsTest` when try to run `verify` task, that's why it took me so many commits🙃 Any idea why this is happening, for the future?

**Review by petertrr — APPROVED** [review]
LGTM

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 1 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt]: No, I've never encountered redeclaration errors :) FWIW, sometimes clean build (`mvn clean install`) or invalidating caches in IDEA helps.

**Pull request merged**